### PR TITLE
annie search logic

### PR DESCRIPTION
**Pull request by yueyiwang**
refactor org search results, sort results by closest distance to search loc, add pagination, add default search behavior

TODO after this PR: 
- test the item_type filter after the donation request endpoint is created
- more thoroughly test some of the lat/lon/radius search cases

### CODE REVIEW

File: backend/helper_zero/views/search.py [hunk 1]
```diff
@@ -19,59 +19,71 @@ def list(self, request):
         lon = request.query_params.get('lon')
         radius = request.query_params.get('radius', 10)
         zipcode = request.query_params.get('zipcode')
+        offset = int(request.query_params.get('offset', 0))
```
Comment (vicky11z) [inline backend/helper_zero/views/search.py]: What alternative besides offset=0 were you thinking the frontend passes in? It's not super clear to me what the behavior of a nonzero offset is

File: backend/helper_zero/views/search.py [hunk 2]
```diff
@@ -19,59 +19,71 @@ def list(self, request):
...
+    if lat and lon:
+        # Order by closest to search location
+        org_search_results.sort(key=lambda x: _get_distance(float(x['lat']), lat, float(x['lon']), lon))
+
+    org_search_results = _paginate_results(org_search_results, offset)
+    return org_search_results
+
+
+def _paginate_results(org_search_results, offset):
+    paginator = Paginator(org_search_results, 20)  # Show 20 orgs per page
```
Comment (vicky11z) [inline backend/helper_zero/views/search.py]: what do you think of allowing the frontend to define how many results they want to show on a page?

Comment (yueyiwang) [reply backend/helper_zero/views/search.py]: yeah i was considering that, but couldn't think of when/why we would need to do that so just put a default in for now; we can fix later or lmk if you feel super strongly

File: backend/helper_zero/views/search.py (hunk 1 above)
Comment (yueyiwang) [reply backend/helper_zero/views/search.py]: assuming that the frontend makes another /search request when a user clicks "view more results", it'll increment offset and send it over so the backend can paginate the results shown

Comment (vicky11z) [reply backend/helper_zero/views/search.py]: ah i see, so it pulls the list of all results again and then grabs the page given the offset? Can you add validation here that offset >= 0 & error if it's false - then can get rid of the check for `page_index > 1` in the pagination function. Think that makes it a little clearer to understand.

File: backend/helper_zero/views/search.py (hunk 2 above)
Comment (vicky11z) [reply backend/helper_zero/views/search.py]: 👍 thinking it'd be useful if we wanted it to be mobile-accessible but not something to worry about for now

File: backend/helper_zero/views/search.py (hunk 1 above)
Comment (yueyiwang) [reply backend/helper_zero/views/search.py]: yeah each page will be a separate search request

i check page_index > 1 to check if we're past the first page of results; if so, i use the previous page index to call the built in django pagination `.has_next()` function https://docs.djangoproject.com/en/3.0/topics/pagination/

Comment (vicky11z) [reply backend/helper_zero/views/search.py]: ah i see what you mean. sounds good then!

Comment (yueyiwang) [reply backend/helper_zero/views/search.py]: added a descriptive comment since it's confusing but i couldn't find a better way thus far (y)

**Pull request merged**